libs/sdk-ui-loaders/src/dashboard/dashboardLoader.tsx

```diff
@@ -272,27 +272,34 @@ export class DashboardLoader implements IDashboardLoader {
 function initializeEmbeddedPlugins(
     ctx: DashboardContext,
     embeddedPlugins: IEmbeddedPlugin[],
-): IDashboardPluginContract_V1[] {
-    return embeddedPlugins.map((embedded) => {
-        const plugin = embedded.factory();
-        plugin.onPluginLoaded?.(ctx, embedded.parameters);
+): Promise<IDashboardPluginContract_V1[]> {
+    const plugins: LoadedPlugin[] = embeddedPlugins.map((embedded) => ({
+        plugin: embedded.factory(),
+        parameters: embedded.parameters,
+    }));
 
-        return plugin;
-    });
+    return initializeLoadedPlugins(ctx, plugins);
 }
 
-function initializeLoadedPlugins(
+async function initializeLoadedPlugins(
     ctx: DashboardContext,
     plugins: LoadedPlugin[],
-): IDashboardPluginContract_V1[] {
+): Promise<IDashboardPluginContract_V1[]> {
     const validPlugins: IDashboardPluginContract_V1[] = [];
 
-    plugins.forEach(({ plugin, parameters }) => {
+    for (const loadedPlugin of plugins) {
+        const { plugin, parameters } = loadedPlugin;
+
         try {
             if (plugin.onPluginLoaded) {
                 // eslint-disable-next-line no-console
                 console.debug(`Calling onPluginLoaded on ${plugin.displayName}...`);
-                plugin.onPluginLoaded(ctx, parameters);
+                const loadPromise = plugin.onPluginLoaded(ctx, parameters);
+
+                // just in case someone forgets to return Promise.resolve()
```

Comment:
I think we can just do
```ts
await plugin.onPluginLoaded(ctx, parameters);
```
you can await anything, even `undefined` so no need for the if. Or am I missing something?

Reply:
fixed